src/routes/posts/+page.server.ts

```diff
@@ -3,18 +3,24 @@ import type { Article } from '$lib/types';
 
 export const load: PageServerLoad = async ({ platform, url }) => {
 	const page = parseInt(url.searchParams.get('page') ?? '0');
+	const category = url.searchParams.get('category')
+		? parseInt(url.searchParams.get('category') as string)
+		: null;
+	const start = url.searchParams.get('start') ?? '2022-12-31';
+	const end = url.searchParams.get('end') ?? new Date().toISOString().split('T')[0];
 
 	const response = await platform?.env.DB.prepare(
 		`SELECT aid FROM articles
-		WHERE images_published_with_alt_text != images_published
+		WHERE images_published_with_alt_text != images_published 
+		${category ? `AND categories LIKE "%${category}%"` : ''}
```

Comment:
I don't _love_ the `LIKE` here. But if it works, it works. Another approach would require changing the categories column into a JSON array and using a JSON query: <https://developers.cloudflare.com/d1/reference/query-json/#expand-arrays-for-in-queries>.

Turning the categories into a JSON array should be trivial: 1) you can keep the TEXT type, and 2) you can probably just prepend and append a bracket to each category entry in a migration file.